Added PodLifecycleSleepActionAllowZero to sig-node-presubmit jobs

sreeram-venkitesh opened the pull request:
This PR adds the e2e test for PodLifecycleSleepActionAllowZero to sig-node-presubmit jobs. This is based on @HirazawaUi's PR https://github.com/kubernetes/test-infra/pull/33687 based on this comment: https://github.com/kubernetes/kubernetes/pull/130621#issuecomment-2708731636.

@HirazawaUi Please take a look!

HirazawaUi commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -3539,3 +3539,57 @@ presubmits:
+      path_alias: k8s.io/release
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20241017-ea6eab1555-master
```
```suggestion
      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20250227-3a13bdd784-master
```

HirazawaUi commented:
/assign @kannon92 
for approval

kannon92 commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -3539,3 +3539,57 @@ presubmits:
           requests:
             cpu: 4
             memory: 6Gi
+  - name: pull-kubernetes-e2e-pod-lifecycle-sleep-action-allow-zero
```
I think  this is using deprecated kubernetetes_e2e.py. Can you update this using kubetest2?

kannon92 commented:
/hold

Please try to use kubetest2 for this.

sreeram-venkitesh commented:
@kannon92 Can you please review https://github.com/kubernetes/test-infra/pull/34483/commits/74f5b1f730029aa9e84dfd08b270b00ab46c5fff? I referred another job in the same file which uses kubetest2 for making this change.

kannon92 commented:
What kind of test is this? e2e or e2e_node?

I see that you picked crio for this one and I'm not sure if that is intendend.

sreeram-venkitesh commented:
This is an e2e test (It comes under e2e/common/node/lifecycle_hook.go). I'm going through sig-node-presubmit.yaml now and all the test are e2e_node ones I'm guessing? Sorry for the confusion, this is my first time doing this.

kannon92 commented:
We could try it!

Please change the image config to use cgroup v2 and I'll approve.

There are other containerd examples for kubetest2 also but if crio works then I think it is fine.

sreeram-venkitesh commented:
Thanks I've updated it! I would like to try running this job locally once before merging it though. I can do this tomorrow and update here. What do you think?

kannon92 commented:
@sreeram-venkitesh have you coordinated with @AxeZhan on this?

You are adding a test for your sub feature but do we have testing for the PodLifecycleSleepAction? We could just add this job to that one?

kannon92 commented:
AFAIK https://github.com/kubernetes/kubernetes/pull/128046 was added to try and graduate this feature.

The test flaked as it was not yet added to a periodic

I think you could add your test to that existing job.

AxeZhan commented on config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml:
```diff
@@ -3539,3 +3539,58 @@ presubmits:
+          - --repo-root=.
+          - --gcp-zone=us-west1-b
+          - --parallelism=1
+          - --focus-regex=\[Feature:PodLifecycleSleepActionAllowZero\]
```
It would be good to take `PodLifecycleSleepAction` with this.

AxeZhan commented:
> I think you could add your test to that existing job.

By `existing job`, do you mean the periodic one got reverted or?

sreeram-venkitesh commented:
> > I think you could add your test to that existing job.
>
> By existing job, do you mean the periodic one got reverted or?

I had the same question! To confirm, is the only change required in `config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml`, where we add `PodLifecycleSleepActionAllowZero` to the `pull-kubernetes-e2e-gce-cos-alpha-features` job like so? 

```diff
- - --test_args=--ginkgo.focus=\[Feature:(WatchList|InPlacePodVerticalScaling|APIServerTracing|SidecarContainers|StorageVersionAPI|PodPreset|ClusterTrustBundle|ClusterTrustBundleProjection|PodLifecycleSleepAction)\] --ginkgo.skip=\[Feature:(SCTPConnectivity|Volumes|Networking-Performance)\]|IPv6|csi-hostpath-v0|\[KubeUp\] --minStartupPods=8
+ - --test_args=--ginkgo.focus=\[Feature:(WatchList|InPlacePodVerticalScaling|APIServerTracing|SidecarContainers|StorageVersionAPI|PodPreset|ClusterTrustBundle|ClusterTrustBundleProjection|PodLifecycleSleepAction|PodLifecycleSleepActionAllowZero)\] --ginkgo.skip=\[Feature:(SCTPConnectivity|Volumes|Networking-Performance)\]|IPv6|csi-hostpath-v0|\[KubeUp\] --minStartupPods=8
```

Or should I keep the new job that I've added in this PR and add `PodLifecycleSleepAction` also to it as commented by @AxeZhan above?

kannon92 commented:
I think its up to you as the feature authors.

You will be the one monitoring and fixing these if issues arise.

Adding to an existing presubmit makes sense.

https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv2-node-e2e-features and the containerd one run the PodLifecycleSleepAction in CI already.

sreeram-venkitesh commented:
I'm fine with having the test run along with the PodLifecycleSleepAction one. 

> https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv2-node-e2e-features and the containerd one run the PodLifecycleSleepAction in CI already.

For this would this change be enough? https://github.com/kubernetes/test-infra/pull/34483/commits/8af7fdeea9c41bc513ad9e8131539818ac4d2122. If yes, I will clean up the rest of the commit.

sreeram-venkitesh commented:
> I'm fine with having the test run along with the PodLifecycleSleepAction one.

I've removed the new job that I've added and have added the feature gate to `pull-kubernetes-e2e-gce-cos-alpha-features`similar to what @AxeZhan did in https://github.com/kubernetes/test-infra/pull/31959. I've also removed the feature gate skip from `ci-crio-cgroupv2-node-e2e-features`. Is this everything we need to make the e2e tests run in CI for our requiements for beta? Thanks!

kannon92 commented:
Thanks for following up on this.

I don’t think you removed that skip from all relevant jobs.

I see https://testgrid.k8s.io/sig-node-cri-o#ci-crio-cgroupv1-node-e2e-features and I think we also have some presubmits for these feature jobs which may need to be updated.

im fine with this as a follow up though.

/lgtm
/approve

you may need other approves for this.

pacoxu commented:
/approve
/assign @andrewsykim @cheftako

ffromani commented:
/hold cancel

per @kannon92 further comments it seems to me the hold no longer applies

mrunalp commented:
Can we clean up the commits here?

kannon92 commented:
> /hold cancel
> 
> per @kannon92 further comments it seems to me the hold no longer applies

I gave my approval for node but I don't have rights for cloud provider.

sreeram-venkitesh commented:
We can merge this now that https://github.com/kubernetes/kubernetes/pull/130621 has been merged.

kannon92 commented:
> We can merge this now that [kubernetes/kubernetes#130621](https://github.com/kubernetes/kubernetes/pull/130621) has been merged.

@andrewsykim @cheftako 

can you please give your blessing?

kannon92 commented:
Sad..  it would be nice to get this PR in.

@dims or @BenTheElder are you able to help here?

BenTheElder commented:
/hold

BenTheElder commented:
/hold cancel
